src/pages/registration/FilesAndLicensePanel.tsx

```diff
@@ -222,27 +224,85 @@ export const FilesAndLicensePanel = ({ uppy }: FilesAndLicensePanelProps) => {
-                                        // Ensure field is set to touched even if it's empty
-                                        setFieldTouched(name);
-                                      }
-                                    }}
-                                    toggleLicenseModal={toggleLicenseModal}
-                                    baseFieldName={`${name}[${associatedFileIndex}]`}
-                                  />
-                                );
+                                        if (associatedArtifactsBeforeRemoval === 1) {
+                                          // Ensure field is set to touched even if it's empty
+                                          setFieldTouched(name);
+                                        }
+                                      }}
+                                      toggleLicenseModal={toggleLicenseModal}
+                                      baseFieldName={`${name}[${associatedFileIndex}]`}
+                                    />
+                                  );
+                                }
+                                return null;
+                              })}
+                            </TableBody>
+                          </Table>
+                        </TableContainer>
+                      </Box>
+                    )}
+
+                    {filesNotToPublish.length > 0 && (
+                      <Box sx={{ display: 'flex', flexDirection: 'column', gap: '1rem', mb: '2rem' }}>
+                        <Typography variant="h2">
+                          {t('registration.files_and_license.files_are_not_published')}
+                        </Typography>
+                        <TableContainer component={Paper}>
+                          <Table sx={alternatingTableRowColor}>
+                            <TableHead>
+                              <TableRow>
+                                <TableCell>{t('common.name')}</TableCell>
+                                <TableCell>{t('registration.files_and_license.size')}</TableCell>
+                                <TableCell>{t('registration.files_and_license.administrative_agreement')}</TableCell>
+                              </TableRow>
+                            </TableHead>
+                            <TableBody>
+                              {files.map((file) => {
+                                const associatedFileIndex = associatedArtifacts.findIndex((artifact) => {
+                                  if (associatedArtifactIsFile(artifact)) {
+                                    const associatedFile = artifact as AssociatedFile;
+                                    return associatedFile.identifier === file.identifier;
+                                  }
+                                  return false;
+                                });
+
+                                if (file.administrativeAgreement) {
```

Comment:
Samme som andre kommentar. Her kan vi mappe gjennom `filesNotToPublish` i stedet for `files`